File: Pillars Management System/pillars-management-system-front/src/app/service/customer.service.ts
import { Router } from '@angular/router';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { CustomerAndPhoneNumbers } from '../data/customerAndPhoneNumbers';
import { Customer } from '../data/customer';

@Injectable({
  providedIn: 'root',
})
export class CustomerService {
  public customersAndPhoneNumbers: Map<number, CustomerAndPhoneNumbers> =
    new Map<number, CustomerAndPhoneNumbers>();

  constructor(private http: HttpClient, private router: Router) {}

  public getCustomersAndPhoneNumbers() {
    return this.http.get(
      'https://localhost:44356/api/customer/getCustomersAndPhoneNumbers'
    );
  }

  public getEmail(email: string) {
    let body = { email: email };

    return this.http.put('https://localhost:44356/api/customer/getEmail', body);
  }

  public addCustomer(newCustomer: Customer) {
    return this.http
      .post('https://localhost:44356/api/customer', newCustomer)
      .subscribe();
  }

  public removeCustomer(cusId: number) {
    return this.http
      .delete(`https://localhost:44356/api/customer/${cusId}`)
      .subscribe();
  }

  public updateCustomer(customer: Customer) {
    return this.http
      .put('https://localhost:44356/api/customer', customer)
      .subscribe();
  }
}

File: Pillars Management System/pillars-management-system-api/Pillars.Sys.Infra/Service/CustomerService.cs
using Pillars.Sys.Core.Data;
using Pillars.Sys.Core.DataTransferObject;
using Pillars.Sys.Core.RepositoryInterface;
using Pillars.Sys.Core.ServiceInterface;
using System;
using System.Collections.Generic;
using System.Text;

namespace Pillars.Sys.Infra.Service
{
    public class CustomerService: ICustomerService
    {
        private readonly ICustomerRepository customerRepository;

        public CustomerService(ICustomerRepository _customerService)
        {
            customerRepository = _customerService;
        }

        public bool AddCustomer(Customer customer)
        {
            return customerRepository.AddCustomer(customer);
        }

        public bool DeleteCustomer(int customerId)
        {
            return customerRepository.DeleteCustomer(customerId);
        }

        public List<Customer> GetCustomers()
        {
            return customerRepository.GetCustomers();
        }

        public bool UpdateCustomer(Customer customer)
        {
            return customerRepository.UpdateCustomer(customer);
        }

        public List<PhoneNumber> GetPhoneNumbersForCustomer(int cus_id)
        {
            return customerRepository.GetPhoneNumbersForCustomer(cus_id);
        }

        public List<CustomerAndPhoneNumbers> GetCustomersAndPhoneNumbers()
        {
            List<CustomerAndPhoneNumbers> customersAndPhoneNumbers = new List<CustomerAndPhoneNumbers>();
            List<Customer> customers = this.GetCustomers();
            
            foreach (Customer customer in customers)
            {
                CustomerAndPhoneNumbers customerAndPhoneNumbers = new CustomerAndPhoneNumbers();
                customerAndPhoneNumbers.Customer = customer;
                customerAndPhoneNumbers.PhoneNumbers = this.GetPhoneNumbersForCustomer(customer.Id);
                customersAndPhoneNumbers.Add(customerAndPhoneNumbers);
            }

            return customersAndPhoneNumbers;
        }

        public Customer GetEmail(string email)
        {
            return this.customerRepository.GetEmail(email);
        }
    }
}
